solutions/usaco-809.mdx

```diff
@@ -2,14 +2,95 @@
 id: usaco-809
 source: USACO Bronze 2018 Febuary
 title: Taming the Herd
-author: Sathvik Chundru
+author: Sathvik Chundru, Aditya Gupta
 ---
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_bronze_feb18.html)
 
 **Time Complexity:** $\mathcal{O}(N)$
 
 <LanguageSection>
+	
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int main() {
+	freopen("taming.in", "r", stdin);
+	freopen("taming.out", "w", stdout);
+	int n;
+	bool valid = true;
+	cin >> n;
+	/*
+	 * A status of 2 means a breakout happened on day n.
+	 * A status of 1 means a breakout did not happen on day n.
+	 * A status of 0 means we are not sure what happened on day n.
+	 */
+	int status[n] = {0};
+	status[0] == 2;  // first day is a special case
+	int log[n];  // log for each day
+	for (int i = 0; i < n; i++) {
+		cin >> log[i];
+	}
+	if (log[0] == 1) {
+		valid = false; // we found a discrepancy
+	} else {
+		status[0] = 2; // the problem says the first day had a breakout
+	}
+	
+	// we go through our log, updating our status array accordingly
+	for (int i = 1; i < n; i++) {
+		if (!valid) break;
+		int curr_n = log[i];
```

Comment:
use a better variable name than ```curr_n``` since having that name doesn't tell me anything